src/explore-education-statistics-frontend/src/services/sitemapService.ts

```diff
@@ -0,0 +1,141 @@
+import methodologyService, {
+  MethodologySitemapSummary,
+} from '@common/services/methodologyService';
+import publicationService, {
+  PublicationSitemapSummary,
+} from '@common/services/publicationService';
+import dataSetService, {
+  DataSetSitemapSummary,
+} from '@common/services/dataSetService';
+import { ISitemapField } from 'next-sitemap';
+
+export default async function getSitemapFields(): Promise<ISitemapField[]> {
+  const methodologies = await methodologyService.getSitemapSummaries();
+  const publications = await publicationService.getSitemapSummaries();
+  const dataSets = await dataSetService.getSitemapSummaries();
+
+  const sitemapFields: ISitemapField[] = [
+    // matches dynamic routes following the patterns:
+    //  methodology/[methodology]
+    ...buildMethodologyRoutes(methodologies),
+
+    // matches dynamic routes following the patterns:
+    //  data-catalogue/[publicationSlug]
+    //  data-tables/[publicationSlug]
+    //  find-statistics/[publication]
+    //  find-statistics/[publication]/data-guidance
+    //  find-statistics/[publication]/prerelease-access-list
+    //  find-statistics/[publication]/[release]/data-guidance
+    //  find-statistics/[publication]/[release]/prerelease-access-list
+    //  data-catalogue/[publicationSlug]/[releaseSlug]
+    //  data-tables/[publicationSlug]/[releaseSlug]
+    //  find-statistics/[publication]/[release]
+    ...buildPublicationRoutes(publications),
+
+    // matches dynamic routes following the patterns:
+    //  data-catalogue/data-set/[dataSetFieldId]
+    ...buildDataSetRoutes(dataSets),
+
+    // Other routes which exist on the site but are excluded from the sitemap by config:
+    //  data-tables/fast-track/[dataBlockParentId]
+    //  data-tables/permalink/[publicationSlug]
+    //  subscriptions/new-subscription/[publicationSlug]
+    //  subscriptions/[publicationSlug]/confirm-subscription/[token]
+    //  subscriptions/[publicationSlug]/confirm-unsubscription/[token]
+  ];
+
+  return sitemapFields;
+}
+
+function buildMethodologyRoutes(
+  methodologies: MethodologySitemapSummary[],
+): ISitemapField[] {
+  return methodologies.map(methodology => ({
+    loc: `${process.env.PUBLIC_URL}methodology/${methodology.slug}`,
+    lastmod: methodology.lastModified
+      ? `${methodology.lastModified}`
+      : undefined,
+  }));
+}
+
+function buildPublicationRoutes(
+  publications: PublicationSitemapSummary[],
+): ISitemapField[] {
+  let fields: ISitemapField[] = [];
+
+  publications.forEach(publication => {
+    fields = fields.concat([
+      {
+        loc: `${process.env.PUBLIC_URL}data-catalogue/${publication.slug}`,
+        lastmod: publication.lastModified
+          ? `${publication.lastModified}`
+          : undefined,
+      },
+      {
+        loc: `${process.env.PUBLIC_URL}data-tables/${publication.slug}`,
+        lastmod: publication.lastModified
+          ? `${publication.lastModified}`
+          : undefined,
+      },
```

Comment:
Do these paths need to be indexed? This probably isn't going to help our SEO much and don't feel like pages that should be indexed and show up in search results IMO.

Might want to check this?